packages/vite/src/plugin.ts

```diff
@@ -0,0 +1,57 @@
+import { resolver } from './resolver.js';
+import { hbs } from './hbs.js';
+import { scripts } from './scripts.js';
+import { templateTag } from './template-tag.js';
+import { optimizeDeps } from './optimize-deps.js';
+import { compatPrebuild } from './build.js';
+import { assets } from './assets.js';
+import { contentFor } from './content-for.js';
+import { babel } from '@rollup/plugin-babel';
+import { type ConfigEnv, type UserConfig } from 'vite';
+
+export const extensions = ['.mjs', '.gjs', '.js', '.mts', '.gts', '.ts', '.hbs', '.json'];
+
+export default function ember() {
+  return [
+    {
+      name: 'vite-plugin-ember',
+      enforce: 'pre',
+      config(_config: UserConfig, env: ConfigEnv) {
+        return {
+          resolve: {
+            extensions,
+          },
+          optimizeDeps: optimizeDeps(),
```

Comment:
if I specify my own optimizeDeps in the userland config, do these two objects get merged together? or should `_config` be spread into `optimizeDeps` here, for every object / array emitted by `optimizeDeps()`